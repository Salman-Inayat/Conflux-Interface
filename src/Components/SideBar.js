import React, { useState, useEffect } from "react";
// import {Grid, Radio, RadioGroup, FormLabel} from '@material-ui/core';
import Grid from "@material-ui/core/Grid";
import FormLabel from "@material-ui/core/FormLabel";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import RadioGroup from "@material-ui/core/RadioGroup";
import Radio from "@material-ui/core/Radio";
import Paper from "@material-ui/core/Paper";
import SankeyChart from "./SankeyChart";
import axios from "axios";

const SideBar = () => {
  const [labelvalue, setLabelValue] = useState("confirm_email");
  const [data, setData] = useState([[]]);
  const [demo, setDemo] = useState();

  const labels = [
    "confirm_email",
    "signup_form",
    "payment_success",
    "product_page",
    "search",
    "landing",
    "input_email",
    "signup_button",
    "input_password",
    "apply_filters",
    "payment_info",
    "view_more_details",
  ];

  // useEffect(() => {
  //   // console.log(data);
  // }, [data]);

  const chartData = [
    ["From", "To", "Weight"],
    ["demo", "signup_form", 4],
    ["demo", "product_page", 5],
    ["demo", "search", 7],

    ["signup_form", "landing", 2],
    ["signup_form", "imput_email", 1],
    ["signup_form", "view_more_details", 1],
    ["product_page", "apply_filters", 1],
    ["product_page", "payment_info", 3],
    ["product_page", "view_more_details", 1],
    ["search", "payment_info", 1],
    ["search", "view_more_details", 1],
    ["search", "payment_success", 2],
    ["search", "landing", 3],
    ["landing", "apply_filters", 1],
    ["landing", "input_password", 3],
    ["payment_info", "input_password", 3],
    ["payment_info", "input_email", 3],
    ["view_more_details", "apply_filters", 1],
  ];

    let data_keys = [[]]
    let data_values = []
    let final_data = []

  function handleChange(event) {
    const clicked_event = event.target.value;
    setLabelValue(clicked_event);
    setDemo("clicked");

    axios
      .get("http://localhost:3001/" + clicked_event)
      .then(function (response) {
        let i;
        let response_data = response.data;

        const parsed_data = JSON.parse(JSON.stringify(response_data));
        data_keys[0].push(Object.keys(parsed_data[0][0]))

        for (i = 0; i < parsed_data[0].length; i++) {
          data_values.push(Object.values(parsed_data[0][i]))
        }

        final_data  = data_keys.concat(data_values)
        // final_data = data_values
        final_data[0] = ["From", "To", "Weight"]
        setData(final_data)

      });
      

    //   function arrayToJSONObject (arr){
    //     var keys = arr[0];
    //     var newArr = arr.slice(1, arr.length);

    //     var formatted = [],
    //     data = newArr,
    //     cols = keys,
    //     l = cols.length;
    //     for (var i=0; i<data.length; i++) {
    //             var d = data[i],
    //                     o = {};
    //             for (var j=0; j<l; j++)
    //                     o[cols[j]] = d[j];
    //             formatted.push(o);
    //     }
    //     return formatted;
    // }

    //   const return_object = arrayToJSONObject(chartData);
    //   console.log(return_object)

    //   axios
    //   .post("http://localhost:3001/demo", return_object)
    //   .then(function (response) {
    //     console.log("posted")
    //   });
  }

  return (
    <Grid container spacing={2}>
      <Grid item md={4} style={{ marginLeft: "20px" }}>
        <Paper>
          <FormLabel style={{ margin: "10px 35px" }} color="primary ">
            Search For
          </FormLabel>
          <RadioGroup
            name="spacing"
            aria-label="spacing"
            value={labelvalue}
            onChange={handleChange}
            column
          >
            {labels.map((value) => (
              <FormControlLabel
                key={value}
                value={value}
                control={<Radio />}
                label={value}
                style={{ margin: "-5px 20px" }}
              />
            ))}
          </RadioGroup>
        </Paper>
      </Grid>
      <Grid item md={7}>
        {demo ? <SankeyChart text={data} /> : <SankeyChart text={chartData} />}
      </Grid>
    </Grid>
  );
};

export default SideBar;
