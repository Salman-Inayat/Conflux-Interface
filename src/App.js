import React from "react";
import SideBar from "./Components/SideBar";
import Grid from "@material-ui/core/Grid";
import Header from "./Components/Header";
import {
  ApolloClient,
  ApolloProvider,
  InMemoryCache,
  HttpLink,
} from "@apollo/client";
// import Customer_Card from "./Components/Customer_Card"

const defaultOptions = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'ignore',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
}

const createApolloClient = () => {
  return new ApolloClient({
    link: new HttpLink({
      uri: "http://localhost:8080/v1/graphql",
    }),
    defaultOptions: defaultOptions,
    cache: new InMemoryCache(),
  });
};

const App = () => {
  const client = createApolloClient();
  return (
    <ApolloProvider client={client}>
      <Grid container spacing={2}>
        <Grid md={12} xs={12}>
          <Header />
        </Grid>
        <Grid md={12}>
          <SideBar />
        </Grid>
        {/* <Grid md={8}>
        <SankeyChart />
        </Grid> */}
        {/* <Grid md={12}>
        <Customer_Card/>
        </Grid> */}
      </Grid>
    </ApolloProvider>
  );
};

export default App;
