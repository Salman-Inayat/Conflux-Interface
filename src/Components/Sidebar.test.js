import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent, cleanup , waitFor} from "@testing-library/react";
import SideBar from "./SideBar";
import { GET_SEARCH, GET_CONFIRM_EMAIL } from "./SideBar";
import TestRenderer from "react-test-renderer";
import { MockedProvider, mockSingleLink } from "@apollo/client/testing";
import { createMockClient } from 'mock-apollo-client';
import { gql, useLazyQuery, InMemoryCache} from "@apollo/client";
import App from "../App"

afterEach(cleanup);

it("Check if any radio button is checked without triggering any event", () => {
  const { getByLabelText } = render(<SideBar />);
  const r_button1 = getByLabelText("landing");
  const r_button2 = getByLabelText("search");
  expect(r_button1).not.toBeChecked();
  expect(r_button2).not.toBeChecked();
});

it("Test if any radio button is checked when a button is clicked", () => {
  const { getByLabelText } = render(<SideBar />);
  const r_button1 = getByLabelText("landing");
  const r_button2 = getByLabelText("search");
  fireEvent.click(r_button1);
  expect(r_button1).toBeChecked();
  expect(r_button2).not.toBeChecked();
});

it("Check value of radio button when changed", () => {
  const { getByLabelText } = render(<SideBar />);
  const payment_info_button = getByLabelText("payment_info");
  fireEvent.change(payment_info_button, { target: { value: "payment_info" } });
  expect(payment_info_button.value).toBe("payment_info");
});

it("should update state on click", () => {
  const setLabelValue = jest.fn();
  const { getByLabelText } = render(<SideBar onChange={setLabelValue} />);
  const landing_button = getByLabelText("landing");
  const handleClick = jest.spyOn(React, "useState");
  handleClick.mockImplementation((labelvalue) => [labelvalue, setLabelValue]);
  fireEvent.change(landing_button);
  expect(setLabelValue).toBeTruthy();
});


const response_data = {
    data: {
      search: [
        {
          To: "signup_button",
          From: "input_password",
          Weight: 3,
        },
        {
          To: "apply_filters",
          From: "input_password",
          Weight: 2,
        },
        {
          To: "signup_button",
          From: "    product_page",
          Weight: 3,
        },
        {
          To: "apply_filters",
          From: "    product_page",
          Weight: 1,
        },
        {
          To: "apply_filters",
          From: "landing",
          Weight: 1,
        },
        {
          To: "search",
          From: "signup_button",
          Weight: 6,
        },
        {
          To: "search",
          From: "apply_filters",
          Weight: 4,
        },
        {
          To: "payment_info",
          From: "search",
          Weight: 4,
        },
        {
          To: "input_email",
          From: "search",
          Weight: 6,
        },
        {
          To: "payment_success",
          From: "payment_info",
          Weight: 4,
        },
        {
          To: "payment_success",
          From: "input_email",
          Weight: 1,
        },
        {
          To: "view_more_details",
          From: "input_email",
          Weight: 3,
        },
        {
          To: "confirm_email",
          From: "input_email",
          Weight: 2,
        },
      ],
    },
  }


it("Renders graphQL query and save snapshot", async () => {
  const mocks = [
    {
      request: {
        query: GET_SEARCH,
      },
      result: response_data
    },
  ];

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SideBar />
    </MockedProvider>
  );

  await waitFor(() => new Promise((res) => setTimeout(res, 0)));

  expect(container).toMatchSnapshot();
});

it("Check graphQL query for error", async () => {
    const mocks = [
      {
        request: {
          query: GET_SEARCH,
        },
        error: new Error('An error occurred'),
      },
    ];
  
    const { findByText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>
    );
  
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const productTag = await findByText("Error");
    expect(productTag).toBeInTheDocument();
  });