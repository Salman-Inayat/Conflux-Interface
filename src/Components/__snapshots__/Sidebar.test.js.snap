// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders graphQL query and save snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3"
      style="margin-left: 50px;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="background-color: rgb(204, 204, 204); padding: 20px;"
      >
        <label
          class="MuiFormLabel-root"
          style="margin: 10px 35px;"
        >
          Select an Event
        </label>
        <div
          aria-label="spacing"
          class="MuiFormGroup-root"
          column="true"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="confirm_email"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              confirm_email
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="search"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              search
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="signup_form"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              signup_form
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="payment_success"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              payment_success
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="product_page"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              product_page
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="landing"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              landing
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="input_email"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              input_email
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="signup_button"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              signup_button
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="input_password"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              input_password
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="apply_filters"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              apply_filters
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="payment_info"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              payment_info
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
            style="margin: -5px 20px; color: black;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-32"
                  name="spacing"
                  type="radio"
                  value="view_more_details"
                />
                <div
                  class="PrivateRadioButtonIcon-root-33"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              view_more_details
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-7"
      style="padding: 50px;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-9"
      >
        <div>
          Loading Chart
        </div>
      </div>
    </div>
    <p
      id="error-div"
      style="display: none;"
    >
      Error
    </p>
  </div>
</div>
`;
